### PR TITLE
hyperscan: add configuration docs

### PR DESCRIPTION
**Pull request by zhxie**
Signed-off-by: Xie Zhihao <zhihao.xie@intel.com>

Commit Message: hyperscan: add configuration docs
Additional Description:
Risk Level: Low
Testing: N/A
Docs Changes: Yes, added in configuration reference
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by zhxie** [comment]
Looks like it cannot be retested. It fails on an example of local ratelimit, which is nothing to do with the PR.

File: docs/root/configuration/other_features/_include/hyperscan_regex_engine.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+static_resources:
```
Comment (htuch) [inline docs/root/configuration/other_features/_include/hyperscan_regex_engine.yaml:1]: @phlax how do we ensure these don't bitrot?

Comment (phlax) [reply docs/root/configuration/other_features/_include/hyperscan_regex_engine.yaml:1]: they are tested in ci both for validity and runnability

**Review by htuch — APPROVED** [review]

**Pull request merged**